netdata: 1.44.3 -> 1.45.3

RaitoBezarius opened the pull request:
## Description of changes

https://github.com/netdata/netdata/releases/tag/v1.45.0

Netdata moved to CMake and required an overhaul of the whole expression. `netdata-go-plugins` has been moved back in the monorepo, leading to a removal of the standalone Go plugin expression.

The eBPF plugin is broken for now.
The patch for disabling the dashboard v2 has been upgraded and sent to upstream again.

This requires a bunch of testing because this upgrade is non-trivial.

I will roll this out on my infrastructure.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

RaitoBezarius committed:
netdata: 1.44.3 -> 1.45.0

https://github.com/netdata/netdata/releases/tag/v1.45.0

Netdata moved to CMake and required an overhaul of the whole expression.
`netdata-go-plugins` has been moved back in the monorepo,
leading to a removal of the standalone Go plugin expression.

The eBPF plugin is broken for now.
The patch for disabling the dashboard v2 has been upgraded and sent to upstream again.
ACLK is correctly rendered optional now.

This adds the network-viewer, logs-management and debugfs plugin
support.

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius commented:
`netdata # [    6.629148] netdata[835]: /nix/store/b1z9yd6zlhqfjzdmz7l2cjfna5sznihj-netdata-1.45.0/bin/netdata: error while loading shared libraries: libprotobuf.so.32: cannot open shared object file: No such file or directory`

Runtime failure.

RaitoBezarius commented:
Now, the problem is that plugins and netdata use the expression to find /var and /etc, we need the equivalent of the previous patches in the new build system.

Izorkin commented:
Check this patch:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8bd91c8d..7e1453a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2138,16 +2138,16 @@ if(NOT CMAKE_INSTALL_PREFIX STREQUAL "")
   string(REGEX REPLACE "/$" "" CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")
 endif()

-set(CACHE_DIR "${CMAKE_INSTALL_PREFIX}/var/cache/netdata")
-set(CONFIG_DIR "${CMAKE_INSTALL_PREFIX}/etc/netdata")
-set(LIBCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")
-set(LOG_DIR "${CMAKE_INSTALL_PREFIX}/var/log/netdata")
-set(PLUGINS_DIR "${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata/plugins.d")
+set(CACHE_DIR "/var/cache/netdata")
+set(CONFIG_DIR "/etc/netdata")
+set(LIBCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/lib/netdata/conf.d")
+set(LOG_DIR "/var/log/netdata")
+set(PLUGINS_DIR "${CMAKE_INSTALL_PREFIX}/libexec/netdata/plugins.d")
 set(VARLIB_DIR "${CMAKE_INSTALL_PREFIX}/var/lib/netdata")

 # A non-default value is only used when building Debian packages (/var/lib/netdata/www)
 if(NOT DEFINED WEB_DIR)
-  set(WEB_DIR "usr/share/netdata/web")
+  set(WEB_DIR "share/netdata/web")
 else()
   string(REGEX REPLACE "^/" "" WEB_DIR "${WEB_DIR}")
 endif()
```

Duplicate `protobuf`:
```
  buildInputs = [ bash curl jemalloc json_c libuv zlib libyaml protobuf ]
...
     ++ lib.optionals (withCloud || withConnPrometheus) [ protobuf ]
```

RaitoBezarius committed:
python3Packages.netdata-pandas: init at 0.0.41

A helper library to pull data from the netdata REST API into a pandas dataframe.

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius committed:
python3Packages.changefinder: init at unstable-2024-03-24

Online changepoint detection.

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius committed:
nixos/netdata: add debugfs, logs-management, network-viewer support

Those require some capabilities.

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius committed:
nixos/netdata: add default programs for netdata (NVMe, WiFi, APCs)

Netdata is zero-config, so we should provide some *default* packages.

If the closure size is a problem for you, reach out to maintainers.

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius committed:
nixos/netdata: introduce `recommendedPythonPackages`

This option furthers the "zero configuration" reputation of netdata by collecting
some Python packages available in nixpkgs and offering them to the module.

It is disabled by default.

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius committed:
nixos/netdata: depends on suid-sgid-wrappers

Netdata is critically dependent on working wrappers, thus, we ensure that the service was successful.

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius committed:
nixos/tests/netdata: use recommended python packages

To maximize the testing surface.

Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius commented:
> Check this patch:
> 
> ```diff
> diff --git a/CMakeLists.txt b/CMakeLists.txt
> index 8bd91c8d..7e1453a8 100644
> --- a/CMakeLists.txt
> +++ b/CMakeLists.txt
> @@ -2138,16 +2138,16 @@ if(NOT CMAKE_INSTALL_PREFIX STREQUAL "")
>    string(REGEX REPLACE "/$" "" CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")
>  endif()
> 
> -set(CACHE_DIR "${CMAKE_INSTALL_PREFIX}/var/cache/netdata")
> -set(CONFIG_DIR "${CMAKE_INSTALL_PREFIX}/etc/netdata")
> -set(LIBCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")
> -set(LOG_DIR "${CMAKE_INSTALL_PREFIX}/var/log/netdata")
> -set(PLUGINS_DIR "${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata/plugins.d")
> +set(CACHE_DIR "/var/cache/netdata")
> +set(CONFIG_DIR "/etc/netdata")
> +set(LIBCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/lib/netdata/conf.d")
> +set(LOG_DIR "/var/log/netdata")
> +set(PLUGINS_DIR "${CMAKE_INSTALL_PREFIX}/libexec/netdata/plugins.d")
>  set(VARLIB_DIR "${CMAKE_INSTALL_PREFIX}/var/lib/netdata")
> 
>  # A non-default value is only used when building Debian packages (/var/lib/netdata/www)
>  if(NOT DEFINED WEB_DIR)
> -  set(WEB_DIR "usr/share/netdata/web")
> +  set(WEB_DIR "share/netdata/web")
>  else()
>    string(REGEX REPLACE "^/" "" WEB_DIR "${WEB_DIR}")
>  endif()
> ```
> 
> Duplicate `protobuf`:
> 
> ```
>   buildInputs = [ bash curl jemalloc json_c libuv zlib libyaml protobuf ]
> ...
>      ++ lib.optionals (withCloud || withConnPrometheus) [ protobuf ]
> ```

I already applied in the form of `substituteInPlace` but thanks.

RaitoBezarius commented:
> Duplicate `protobuf`:
> 
> ```
>   buildInputs = [ bash curl jemalloc json_c libuv zlib libyaml protobuf ]
> ...
>      ++ lib.optionals (withCloud || withConnPrometheus) [ protobuf ]
> ```

Fixed.

RaitoBezarius commented:
Deployed on my parent coordinator successfully.

Izorkin commented:
Replace to?
```
    # Prevent the path to be caught into the Nix store path.
    substituteInPlace CMakeLists.txt \
      --replace-fail 'set(CACHE_DIR "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(CACHE_DIR "/var/cache/netdata")' \
      --replace-fail 'set(CONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(CONFIG_DIR "/etc/netdata")' \
...
```

RaitoBezarius commented:
> Replace to?
> 
> ```
>     # Prevent the path to be caught into the Nix store path.
>     substituteInPlace CMakeLists.txt \
>       --replace-fail 'set(CACHE_DIR "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(CACHE_DIR "/var/cache/netdata")' \
>       --replace-fail 'set(CONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(CONFIG_DIR "/etc/netdata")' \
> ...
> ```

Can you push that to the PR or send me a patch I can apply? Thanks. I already spent too much time on this.

Tested on my whole infrastructure successfully.

Izorkin commented:
Patch:
```diff
diff --git a/pkgs/tools/system/netdata/default.nix b/pkgs/tools/system/netdata/default.nix
index 2f7dea040835..0ed72b26d451 100644
--- a/pkgs/tools/system/netdata/default.nix
+++ b/pkgs/tools/system/netdata/default.nix
@@ -127,45 +127,19 @@ stdenv.mkDerivation rec {

     # Prevent the path to be caught into the Nix store path.
     substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(CACHE_DIR "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(CACHE_DIR "/var/cache/netdata")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(CONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(CONFIG_DIR "/etc/netdata")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(LIBCONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")' 'set(LIBCONFIG_DIR "${placeholder "out"}/share/netdata/conf.d")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(LOG_DIR "''${CMAKE_INSTALL_PREFIX}/var/log/netdata")' 'set(LOG_DIR "/var/log/netdata")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(PLUGINS_DIR "''${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata/plugins.d")' 'set(PLUGINS_DIR "${placeholder "out"}/libexec/netdata/plugins.d")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(VARLIB_DIR "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata")' 'set(VARLIB_DIR "/var/lib/netdata")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(pkglibexecdir_POST "''${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata")' 'set(pkglibexecdir_POST "${placeholder "out"}/libexec/netdata")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(localstatedir_POST "''${CMAKE_INSTALL_PREFIX}/var")' 'set(localstatedir_POST "/var")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(sbindir_POST "''${CMAKE_INSTALL_PREFIX}/usr/sbin")' 'set(sbindir_POST "${placeholder "out"}/bin")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(configdir_POST "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(configdir_POST "/etc/netdata")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(libconfigdir_POST "''${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")' 'set(libconfigdir_POST "${placeholder "out"}/share/netdata/conf.d")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(cachedir_POST "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(libconfigdir_POST "/var/cache/netdata")'
-
-    substituteInPlace CMakeLists.txt \
-      --replace-fail 'set(registrydir_POST "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata/registry")' 'set(registrydir_POST "/var/lib/netdata/registry")'
-
-    substituteInPlace CMakeLists.txt \
+      --replace-fail 'set(CACHE_DIR "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(CACHE_DIR "/var/cache/netdata")' \
+      --replace-fail 'set(CONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(CONFIG_DIR "/etc/netdata")' \
+      --replace-fail 'set(LIBCONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")' 'set(LIBCONFIG_DIR "${placeholder "out"}/share/netdata/conf.d")' \
+      --replace-fail 'set(LOG_DIR "''${CMAKE_INSTALL_PREFIX}/var/log/netdata")' 'set(LOG_DIR "/var/log/netdata")' \
+      --replace-fail 'set(PLUGINS_DIR "''${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata/plugins.d")' 'set(PLUGINS_DIR "${placeholder "out"}/libexec/netdata/plugins.d")' \
+      --replace-fail 'set(VARLIB_DIR "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata")' 'set(VARLIB_DIR "/var/lib/netdata")' \
+      --replace-fail 'set(pkglibexecdir_POST "''${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata")' 'set(pkglibexecdir_POST "${placeholder "out"}/libexec/netdata")' \
+      --replace-fail 'set(localstatedir_POST "''${CMAKE_INSTALL_PREFIX}/var")' 'set(localstatedir_POST "/var")' \
+      --replace-fail 'set(sbindir_POST "''${CMAKE_INSTALL_PREFIX}/usr/sbin")' 'set(sbindir_POST "${placeholder "out"}/bin")' \
+      --replace-fail 'set(configdir_POST "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(configdir_POST "/etc/netdata")' \
+      --replace-fail 'set(libconfigdir_POST "''${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")' 'set(libconfigdir_POST "${placeholder "out"}/share/netdata/conf.d")' \
+      --replace-fail 'set(cachedir_POST "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(libconfigdir_POST "/var/cache/netdata")' \
+      --replace-fail 'set(registrydir_POST "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata/registry")' 'set(registrydir_POST "/var/lib/netdata/registry")' \
       --replace-fail 'set(varlibdir_POST "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata")' 'set(varlibdir_POST "/var/lib/netdata")'
   '';

```

RaitoBezarius commented:
> Patch:
> 
> ```diff
> diff --git a/pkgs/tools/system/netdata/default.nix b/pkgs/tools/system/netdata/default.nix
> index 2f7dea040835..0ed72b26d451 100644
> --- a/pkgs/tools/system/netdata/default.nix
> +++ b/pkgs/tools/system/netdata/default.nix
> @@ -127,45 +127,19 @@ stdenv.mkDerivation rec {
> 
>      # Prevent the path to be caught into the Nix store path.
>      substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(CACHE_DIR "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(CACHE_DIR "/var/cache/netdata")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(CONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(CONFIG_DIR "/etc/netdata")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(LIBCONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")' 'set(LIBCONFIG_DIR "${placeholder "out"}/share/netdata/conf.d")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(LOG_DIR "''${CMAKE_INSTALL_PREFIX}/var/log/netdata")' 'set(LOG_DIR "/var/log/netdata")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(PLUGINS_DIR "''${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata/plugins.d")' 'set(PLUGINS_DIR "${placeholder "out"}/libexec/netdata/plugins.d")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(VARLIB_DIR "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata")' 'set(VARLIB_DIR "/var/lib/netdata")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(pkglibexecdir_POST "''${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata")' 'set(pkglibexecdir_POST "${placeholder "out"}/libexec/netdata")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(localstatedir_POST "''${CMAKE_INSTALL_PREFIX}/var")' 'set(localstatedir_POST "/var")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(sbindir_POST "''${CMAKE_INSTALL_PREFIX}/usr/sbin")' 'set(sbindir_POST "${placeholder "out"}/bin")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(configdir_POST "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(configdir_POST "/etc/netdata")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(libconfigdir_POST "''${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")' 'set(libconfigdir_POST "${placeholder "out"}/share/netdata/conf.d")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(cachedir_POST "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(libconfigdir_POST "/var/cache/netdata")'
> -
> -    substituteInPlace CMakeLists.txt \
> -      --replace-fail 'set(registrydir_POST "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata/registry")' 'set(registrydir_POST "/var/lib/netdata/registry")'
> -
> -    substituteInPlace CMakeLists.txt \
> +      --replace-fail 'set(CACHE_DIR "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(CACHE_DIR "/var/cache/netdata")' \
> +      --replace-fail 'set(CONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(CONFIG_DIR "/etc/netdata")' \
> +      --replace-fail 'set(LIBCONFIG_DIR "''${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")' 'set(LIBCONFIG_DIR "${placeholder "out"}/share/netdata/conf.d")' \
> +      --replace-fail 'set(LOG_DIR "''${CMAKE_INSTALL_PREFIX}/var/log/netdata")' 'set(LOG_DIR "/var/log/netdata")' \
> +      --replace-fail 'set(PLUGINS_DIR "''${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata/plugins.d")' 'set(PLUGINS_DIR "${placeholder "out"}/libexec/netdata/plugins.d")' \
> +      --replace-fail 'set(VARLIB_DIR "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata")' 'set(VARLIB_DIR "/var/lib/netdata")' \
> +      --replace-fail 'set(pkglibexecdir_POST "''${CMAKE_INSTALL_PREFIX}/usr/libexec/netdata")' 'set(pkglibexecdir_POST "${placeholder "out"}/libexec/netdata")' \
> +      --replace-fail 'set(localstatedir_POST "''${CMAKE_INSTALL_PREFIX}/var")' 'set(localstatedir_POST "/var")' \
> +      --replace-fail 'set(sbindir_POST "''${CMAKE_INSTALL_PREFIX}/usr/sbin")' 'set(sbindir_POST "${placeholder "out"}/bin")' \
> +      --replace-fail 'set(configdir_POST "''${CMAKE_INSTALL_PREFIX}/etc/netdata")' 'set(configdir_POST "/etc/netdata")' \
> +      --replace-fail 'set(libconfigdir_POST "''${CMAKE_INSTALL_PREFIX}/usr/lib/netdata/conf.d")' 'set(libconfigdir_POST "${placeholder "out"}/share/netdata/conf.d")' \
> +      --replace-fail 'set(cachedir_POST "''${CMAKE_INSTALL_PREFIX}/var/cache/netdata")' 'set(libconfigdir_POST "/var/cache/netdata")' \
> +      --replace-fail 'set(registrydir_POST "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata/registry")' 'set(registrydir_POST "/var/lib/netdata/registry")' \
>        --replace-fail 'set(varlibdir_POST "''${CMAKE_INSTALL_PREFIX}/var/lib/netdata")' 'set(varlibdir_POST "/var/lib/netdata")'
>    '';
> ```

Applied, thanks.

Izorkin commented:
Error build without IPMI support:
```
-- Checking for module 'libipmimonitoring'
--   No package 'libipmimonitoring' found
CMake Error at /nix/store/17r6ld906midfv8y7997fd56s7a87vrg-cmake-3.28.3/share/cmake-3.28/Modules/FindPkgConfig.cmake:619 (message):
  The following required packages were not found:

   - libipmimonitoring

Call Stack (most recent call first):
  /nix/store/17r6ld906midfv8y7997fd56s7a87vrg-cmake-3.28.3/share/cmake-3.28/Modules/FindPkgConfig.cmake:841 (_pkg_check_modules_internal)
  CMakeLists.txt:1713 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
```

Izorkin commented:
Patch:
```
diff --git a/pkgs/tools/system/netdata/default.nix b/pkgs/tools/system/netdata/default.nix
index 0ed72b26d451..52ca2c11a2bd 100644
--- a/pkgs/tools/system/netdata/default.nix
+++ b/pkgs/tools/system/netdata/default.nix
@@ -150,6 +150,7 @@ stdenv.mkDerivation rec {
     (lib.cmakeBool "ENABLE_ACLK" withCloud)
     (lib.cmakeBool "ENABLE_DASHBOARD_V2" withCloudUi)
     (lib.cmakeBool "ENABLE_DBENGINE" withDBengine)
+    (lib.cmakeBool "ENABLE_PLUGIN_FREEIPMI" withIpmi)
     (lib.cmakeBool "ENABLE_PLUGIN_SYSTEMD_JOURNAL" withSystemdJournal)
     (lib.cmakeBool "ENABLE_PLUGIN_NETWORK_VIEWER" withNetworkViewer)
     (lib.cmakeBool "ENABLE_PLUGIN_EBPF" withEbpf)
```
Next error - web interface does not open:
```
File does not exist, or is not accessible: /nix/store/mijj1rzqqnhpws4y6abf8qa7rs3q1ihy-netdata-1.45.0/share/netdata/web/v2
```

wmertens committed:
netdata: 1.45.0 -> 1.45.3

https://github.com/netdata/netdata/releases/tag/v1.45.3

wmertens commented:
I took the liberty of updating the branch to 1.45.3. It's building now, but 1.45.0 with cloud support already deployed without an issue for me.

wmertens commented:
Ah sorry, since you had said "Can you push that to the PR" I wrongly assumed it was fine. I'll ask next time :)

wmertens commented:
oh - I didn't force push BTW

RaitoBezarius commented:
> oh - I didn't force push BTW

Right, indeed

wmertens committed:
nixos/netdata: remove old lib.mdDoc calls

wmertens commented:
I removed the lib.mdDoc calls so the tests will pass

wmertens commented:
It seems to be working well for me, I'd say the only blocker is the use without cloud, which needs the web interface not to open v2?

pedorich-n commented:
I am having a couple of issues collecting `podman` (running in rootful mode) metrics:
1. `cgroup-name.sh` requires `jq` and `podman` to use podman's socket API: 
```
netdata[988432]: /nix/store/b200312w0w4lrv1ss0g8y1kic1jmqm41-netdata-1.45.3/libexec/netdata/plugins.d/cgroup-name.sh: line 132: podman: command not found
...
cgroup-name.sh[988461]: Can't find jq command line tool. jq is required for netdata to retrieve container name using /run/podman/podman. sock API, falling back to docker ps
```
Currently can be solved by adding `jq` and `podman` to `systemd.services.netdata.path`, like in https://github.com/NixOS/nixpkgs/pull/274577

2. When running from the non-root user (like `netdata`) `cgroup-network` throws a bunch of permission issues trying to collect info about podman containers:
```
Apr 28 22:42:29 machine cgroup-network[988446]: running: exec /nix/store/b200312w0w4lrv1ss0g8y1kic1jmqm41-netdata-1.45.3/libexec/netdata/plugins.d/cgroup-network-helper.sh --cgroup '/sys/fs/cgroup/machine.slice/libpod-f1aa7ace2f0e23f897e27f2dde7a6e81d66bde5f399c85dd4509fb271b5ba748.scope'
Apr 28 22:42:29 machine netdata[988447]: /nix/store/i0w18x1kghg06jmsavj37b9irblxb9yk-set-environment: line 14: warning: setlocale: LC_MESSAGES: cannot change locale (en_US.UTF-8): No such file or directory
Apr 28 22:42:29 machine netdata[988447]: /nix/store/i0w18x1kghg06jmsavj37b9irblxb9yk-set-environment: line 17: warning: setlocale: LC_NUMERIC: cannot change locale (en_US.UTF-8): No such file or directory
Apr 28 22:42:29 machine netdata[988447]: /nix/store/i0w18x1kghg06jmsavj37b9irblxb9yk-set-environment: line 19: warning: setlocale: LC_TIME: cannot change locale (en_GB.UTF-8): No such file or directory
Apr 28 22:42:29 machine .cgroup-network-helper.sh-wrapped[988451]: searching for network interfaces of cgroup '/sys/fs/cgroup/machine.slice/libpod-f1aa7ace2f0e23f897e27f2dde7a6e81d66bde5f399c85dd4509fb271b5ba748.scope'
Apr 28 22:42:29 machine cgroup-network[988446]: child pid 988447 exited with code 1.
Apr 28 22:42:29 machine cgroup-network[988446]: Cannot open proc_pid_fd() file '/proc/24739/ns/net'
Apr 28 22:42:29 machine cgroup-network[988446]: Cannot open proc_pid_fd() file '/proc/24739/ns/pid'
Apr 28 22:42:29 machine cgroup-network[988446]: Cannot open proc_pid_fd() file '/proc/24739/ns/mnt'
Apr 28 22:42:29 machine cgroup-network[988446]: Cannot open proc_pid_fd() file '/proc/24739/root'
Apr 28 22:42:29 machine cgroup-network[988446]: Cannot open proc_pid_fd() file '/proc/24739/cwd'
Apr 28 22:42:29 machine netdata[985530]: child pid 988446 exited with code 1.
``` 
I tried adding different capabilities to `cgroup-network`, but I have no idea what I'm doing so the result was more or less the same.


Would it be possible to look into those issues in this PR?

RaitoBezarius commented:
I don't really use containerization technologies, but I am willing to accept patches to fix the second issue, the first, I can definitely add `jq`.

pedorich-n commented:
For `jq`: wouldn't it be better to add another `wrapProgram` to the `netdata` package rather than modify the service?

For the second one: I am now reading more and it might not be an issue at all. 
See https://github.com/netdata/netdata/issues/16244#issuecomment-1771546675 & https://github.com/netdata/netdata/discussions/16105#discussioncomment-7261968. But I don't have the `net` section in "Containers & VMs" so I think it is an issue.
Anyway, it seems this issue runs deeper, so it's better to explore it separately.

pedorich-n commented:
Also proposal: add `python` option to the service instead of using `pkgs.python3`: https://github.com/NixOS/nixpkgs/blob/f94fc9012a4791afb320e5758d0b8f566d412177/nixos/modules/services/monitoring/netdata.nix#L246

I am now using this service as drop-in replacement like this:
```nix
  disabledModules = [ "services/monitoring/netdata.nix" ];
  imports = [ "${inputs.nixpkgs-netdata-45}/nixos/modules/services/monitoring/netdata.nix" ];
```
and if I enable the option `python.recommendedPythonPackages` it fails to build with 
```
       error: attribute 'netdata-pandas' missing

       at /nix/store/2gf6w16qjmr1hhwy1x0ln2h98cj9cvqb-source/nixos/modules/services/monitoring/netdata.nix:219:7:

          218|       ps.python-ldap
          219|       ps.netdata-pandas
             |       ^
          220|       ps.changefinder
```

I assume because it tries to get the package from my global `pkgs` which is set to `nixos-23.11`.
With an option, I could overwrite to use the python package from this branch too. 

I can work on it, let me know what you think.

RaitoBezarius commented:
> For `jq`: wouldn't it be better to add another `wrapProgram` to the `netdata` package rather than modify the service?

Hmm, we already have a bunch of things we have in systemd's unit path, I'm not sure what does that buy us? Putting it in the netdata package means rebuilding the package, putting it in the service means rebuilding the service unit.

pedorich-n commented:
> Hmm, we already have a bunch of things we have in systemd's unit path, I'm not sure what does that buy us? Putting it in the netdata package means rebuilding the package, putting it in the service means rebuilding the service unit.

I agree. It just feels weird that we set some runtime dependencies in the package and many others in the service. The package without service is not self-sufficient.
Personally, I think something `pkgs.netdata.withRuntimeDepedencies` would make sense. It would create another derivation wrapping the `netdata` package with extra apps in the `PATH`. 
But I don't know if that's in the spirit of NixOS and if it is, then it's probably better done in a separate PR anyway.

pedorich-n commented:
Here's the patch for `python.package` if you think it's worth adding: https://github.com/pedorich-n/nixpkgs/commit/f770f94c3a111cd04cdcc8b851b69237de68891e.patch

And changes from @rhoriguchi adding `jq` can be easily applied after that too: https://github.com/NixOS/nixpkgs/commit/93543f31db9e485c2fe9f2d66b49c1cc1937319b

SuperSandro2000 commented on pkgs/development/python-modules/changefinder/default.nix:
```diff
@@ -0,0 +1,44 @@
+, nose
+}:
+
+buildPythonPackage rec {
```
```suggestion
buildPythonPackage {
```

SuperSandro2000 commented on pkgs/development/python-modules/netdata-pandas/default.nix:
```diff
@@ -0,0 +1,44 @@
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/changefinder/default.nix:
```diff
@@ -0,0 +1,44 @@
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
```
Not sure where this gets recommended but to my knowledge we don't need to add wheel here.


```suggestion
```

SuperSandro2000 reviewed (commented):
just some small notes, mostly on the python packages

pedorich-n commented:
Hey @RaitoBezarius! 
Is there anything we can help you with to get this PR merged? 
It's probably better to just make sure it works and is backward compatible within the scope of this PR and I can open another PR later, with the contributions I suggested.

RaitoBezarius commented:
Yeah, the current governance crisis force me to pause almost all my contributions right now. Feel free to send me a patch to apply to this PR.

rhoriguchi commented:
Content of #274577, PR can be closed if you add this.
 
```diff
diff --git a/nixos/modules/services/monitoring/netdata.nix b/nixos/modules/services/monitoring/netdata.nix
index 00bdd9fc..3c3d425a 100644
--- a/nixos/modules/services/monitoring/netdata.nix
+++ b/nixos/modules/services/monitoring/netdata.nix
@@ -156,7 +156,9 @@ in {
       wantedBy = [ "multi-user.target" ];
       path = (with pkgs; [ curl gawk iproute2 which ])
         ++ lib.optional cfg.python.enable (pkgs.python3.withPackages cfg.python.extraPackages)
-        ++ lib.optional config.virtualisation.libvirtd.enable (config.virtualisation.libvirtd.package);
+        ++ lib.optional config.virtualisation.libvirtd.enable config.virtualisation.libvirtd.package
+        ++ lib.optional config.virtualisation.docker.enable config.virtualisation.docker.package
+        ++ lib.optionals config.virtualisation.podman.enable [ pkgs.jq config.virtualisation.podman.package ];
       environment = {
         PYTHONPATH = "${cfg.package}/libexec/netdata/python.d/python_modules";
       } // lib.optionalAttrs (!cfg.enableAnalyticsReporting) {
@@ -263,6 +265,8 @@ in {
       ${defaultUser} = {
         group = defaultUser;
         isSystemUser = true;
+        extraGroups = lib.optional config.virtualisation.docker.enable "docker"
+          ++ lib.optional config.virtualisation.podman.enable "podman";
       };
     };
```

---

Add [Wireguard](https://learn.netdata.cloud/docs/collecting-metrics/vpns/wireguard) support
> This collector requires the CAP_NET_ADMIN capability, but it is set automatically during installation, so no manual configuration is needed.

```diff
diff --git a/nixos/modules/services/monitoring/netdata.nix b/nixos/modules/services/monitoring/netdata.nix
index 00bdd9fc..919458f5 100644
--- a/nixos/modules/services/monitoring/netdata.nix
+++ b/nixos/modules/services/monitoring/netdata.nix
@@ -187,6 +187,8 @@ in {
         # Configuration directory and mode
         ConfigurationDirectory = "netdata";
         ConfigurationDirectoryMode = "0755";
+        # AmbientCapabilities
+        AmbientCapabilities = lib.optional config.networking.wireguard.enable "CAP_NET_ADMIN";
         # Capabilities
         CapabilityBoundingSet = [
           "CAP_DAC_OVERRIDE"      # is required for freeipmi and slabinfo plugins
@@ -199,7 +201,7 @@ in {
           "CAP_NET_RAW"           # is required for fping app
           "CAP_SYS_CHROOT"        # is required for cgroups plugin
           "CAP_SETUID"            # is required for cgroups and cgroups-network plugins
-        ];
+        ] ++ lib.optional config.networking.wireguard.enable "CAP_NET_ADMIN";
         # Sandboxing
         ProtectSystem = "full";
         ProtectHome = "read-only";
```

rhoriguchi commented:
Some other strange issue I found while testing this is, that when you try to run a plugin manually to debug something it uses the wrong config path `/nix/store/etc/netdata` and `/nix/store/usr/lib/netdata/conf.d`. You get this behavior with any manual execution.

```console
> sudo -u netdata /nix/store/65d3bnd96b0s5jynnxbrj5py3f44k4b3-netdata-1.45.0/libexec/netdata/plugins.d/go.d.plugin -d -m windows
DBG godplugin/main.go:145 plugin: name=go.d, version=1.45.0.0 component=agent
DBG godplugin/main.go:147 current user: name=netdata, uid=991 component=agent
INF godplugin/main.go:151 env HTTP_PROXY '', HTTPS_PROXY '' component=agent
INF agent/agent.go:145 instance is started component=agent
INF agent/setup.go:23 loading config file component=agent
DBG agent/setup.go:31 looking for 'go.d.conf' in [/nix/store/etc/netdata /nix/store/usr/lib/netdata/conf.d] component=agent
WRN agent/setup.go:35 couldn't find config, will use defaults component=agent
INF agent/agent.go:149 using config: enabled 'true', default_run 'true', max_procs '0' component=agent
INF agent/setup.go:50 loading modules component=agent
INF agent/setup.go:73 enabled/registered modules: 1/77 component=agent
INF agent/setup.go:79 building discovery config component=agent
DBG agent/setup.go:114 looking for 'wmi.conf' in [/nix/store/etc/netdata/go.d /nix/store/usr/lib/netdata/conf.d/go.d] component=agent
DBG agent/setup.go:126 looking for 'windows.conf' in [/nix/store/etc/netdata/go.d /nix/store/usr/lib/netdata/conf.d/go.d] component=agent
INF agent/setup.go:139 couldn't find 'windows' module config, will use default config component=agent
INF agent/setup.go:147 dummy/read/watch paths: 1/0/0 component=agent
INF discovery/manager.go:116 registered discoverers: [dummy discovery service discovery] component="discovery manager"
DBG agent/setup.go:165 looking for 'vnodes/' in [/nix/store/etc/netdata /nix/store/usr/lib/netdata/conf.d] component=agent
INF discovery/manager.go:61 instance is started component="discovery manager"
INF sd/sd.go:66 instance is started component="service discovery"
INF functions/manager.go:50 instance is started component="functions manager"
INF jobmgr/manager.go:87 instance is started component="job manager"
INF dummy/discovery.go:45 instance is started component=discovery discoverer=dummy
DBG functions/ext.go:15 registering function 'config' component="functions manager"
INF dummy/discovery.go:46 instance is stopped component=discovery discoverer=dummy
CONFIG go.d:collector:windows create accepted template /collectors/jobs internal 'internal' 'add schema enable disable test' 0x0000 0x0000

DBG jobmgr/manager.go:122 received configs: 1/+1/-0 ('internal') component="job manager"
CONFIG go.d:collector:windows:windows create accepted job /collectors/jobs stock 'internal' 'schema get enable disable update restart test' 0x0000 0x0000

DBG jobmgr/manager.go:293 creating windows[windows] job, config: map[__provider__:dummy __source__:internal __source_type__:stock autodetection_retry:0 module:windows name:windows priority:70000 update_every:5] component="job manager"
ERR windows/windows.go:123 config validation: 'url' is not set collector=windows job=windows
ERR module/job.go:231 init failed collector=windows job=windows
CONFIG go.d:collector:windows:windows delete

^CINF agent/agent.go:113 received interrupt signal (2). Terminating... component=agent
INF functions/manager.go:51 instance is stopped component="functions manager"
INF discovery/manager.go:62 instance is stopped component="discovery manager"
INF jobmgr/manager.go:88 instance is stopped component="job manager"
INF agent/agent.go:146 instance is stopped component=agent
```

pedorich-n committed:
netdata: 1.45.3 -> 1.45.4

pedorich-n commented:
Hey @RaitoBezarius!
Here's the patch addressing @SuperSandro2000 's comments: https://github.com/pedorich-n/nixpkgs/commit/d024426b816bc557ca5fb67725d25458ad7076a4.patch

And here's one bumping netdata to 1.45.4: https://github.com/pedorich-n/nixpkgs/commit/5c81b897fae2f8e3cc519aff68d2c7f3602f9336.patch

Hope that helps.

pedorich-n commented:
@rhoriguchi I think this code causes the issue: 
https://github.com/netdata/netdata/blob/d42b9c6d64be80ec77472e40276ab2f590374fa0/src/go/collectors/go.d.plugin/cmd/godplugin/main.go#L48-L56 and/or
https://github.com/netdata/netdata/blob/d42b9c6d64be80ec77472e40276ab2f590374fa0/src/go/collectors/go.d.plugin/cmd/godplugin/main.go#L73-L81

We could patch those to use an absolute path, but I don't know what consequences it might lead to.
In the meantime, you could try running the collector with `-c /etc/netdata/conf.d/`

rhoriguchi commented:
@pedorich-n nice find seems to be an upstream issue `-c /etc/netdata` fixes that.

rhoriguchi commented:
Here is a patch for the issue mentioned in https://github.com/NixOS/nixpkgs/pull/298641#issuecomment-2103686549

```diff
diff --git a/pkgs/tools/system/netdata/default.nix b/pkgs/tools/system/netdata/default.nix
index 3c025247..26227fa3 100644
--- a/pkgs/tools/system/netdata/default.nix
+++ b/pkgs/tools/system/netdata/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, callPackage, fetchFromGitHub, autoreconfHook, pkg-config, makeWrapper
+{ lib, stdenv, callPackage, fetchFromGitHub, fetchpatch, autoreconfHook, pkg-config, makeWrapper
 , CoreFoundation, IOKit, libossp_uuid
 , nixosTests
 , curl, libcap, libuuid, lm_sensors, zlib
@@ -47,6 +47,13 @@ in stdenv.mkDerivation rec {
     # Therefore we put it into `/run/netdata`, which is owned
     # by netdata only.
     ./ipc-socket-in-run.patch
+    # Can be removed once > v1.45.4 is released
+    # https://github.com/netdata/netdata/pull/17661
+    (fetchpatch {
+      name = "fix-rust-1.54-build.patch";
+      url = "https://patch-diff.githubusercontent.com/raw/netdata/netdata/pull/17661.patch";
+      sha256 = "sha256-j+mrwkibQio2KO8UnV7sxzCoHmkcsalHNzP+YvrRz74=";
+    })
   ];

   NIX_CFLAGS_COMPILE = optionalString withDebug "-O1 -ggdb -DNETDATA_INTERNAL_CHECKS=1";
   ```

pedorich-n commented:
Hi, @RaitoBezarius !
Will you have a chance to get back to this PR anytime soon? Having this PR with the new build system merged would unblock us from working on other changes.

RaitoBezarius committed:
nixos/netdata: add docker and podman support

Podman requires `jq`.

Change-Id: Iad2abe1ccf92738da82384f30503f3be6fb11ca7
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius committed:
nixos/netdata: add wireguard support

CAP_NET_ADMIN is required.

Change-Id: I8559e50ccf2d34a4bc7c8f4f2aeafb7771fe751b
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius committed:
netdata: workaround for go.d plugins directory accesses

Change-Id: I1731e8f3d2c53dfed8d1be3f74f9219f7cf84a34
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius commented:
I upgraded this branch with all your patches, thank you.

RaitoBezarius committed:
netdata: drop maintenance from raitobezarius

Change-Id: If98b9f43825ca4f9a4bd80ad6e0a4fe16c7368dc
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius commented:
This is my final contribution for the netdata packaging in nixpkgs, I will leave it to others to send a PR to add themselves back. As of now, the package is orphaned and I will not contribute the 1.46.0 bump, released some hours ago.

Izorkin commented:
Error:
```
- Checking for module 'libipmimonitoring'
--   No package 'libipmimonitoring' found
CMake Error at /nix/store/rx7cp0slp68fdkgfq7wb6i2sinkn6xr4-cmake-3.29.2/share/cmake-3.29/Modules/FindPkgConfig.cmake:634 (message):
  The following required packages were not found:

   - libipmimonitoring

Call Stack (most recent call first):
  /nix/store/rx7cp0slp68fdkgfq7wb6i2sinkn6xr4-cmake-3.29.2/share/cmake-3.29/Modules/FindPkgConfig.cmake:862 (_pkg_check_modules_internal)
  CMakeLists.txt:1727 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
```
Patch:
```diff
diff --git a/pkgs/tools/system/netdata/default.nix b/pkgs/tools/system/netdata/default.nix
index 7fae2c9c8247..617f40f2bca8 100644
--- a/pkgs/tools/system/netdata/default.nix
+++ b/pkgs/tools/system/netdata/default.nix
@@ -149,6 +149,7 @@ stdenv.mkDerivation rec {
     (lib.cmakeBool "ENABLE_ACLK" withCloud)
     (lib.cmakeBool "ENABLE_DASHBOARD_V2" withCloudUi)
     (lib.cmakeBool "ENABLE_DBENGINE" withDBengine)
+    (lib.cmakeBool "ENABLE_PLUGIN_FREEIPMI" withIpmi)
     (lib.cmakeBool "ENABLE_PLUGIN_SYSTEMD_JOURNAL" withSystemdJournal)
     (lib.cmakeBool "ENABLE_PLUGIN_NETWORK_VIEWER" withNetworkViewer)
     (lib.cmakeBool "ENABLE_PLUGIN_EBPF" withEbpf)

```

RaitoBezarius commented:
> ```diff
> (lib.cmakeBool "ENABLE_PLUGIN_FREEIPMI" withIpmi)
> ```

Done.

Izorkin commented:
Patch ro enable v1 UI if disabled CloudUI:
```diff
diff --git a/pkgs/tools/system/netdata/default.nix b/pkgs/tools/system/netdata/default.nix
index 7fae2c9c8247..9ddba232de19 100644
--- a/pkgs/tools/system/netdata/default.nix
+++ b/pkgs/tools/system/netdata/default.nix
@@ -113,6 +113,10 @@ stdenv.mkDerivation rec {
       mv $out/libexec/netdata/plugins.d/network-viewer.plugin \
          $out/libexec/netdata/plugins.d/network-viewer.plugin.org
     ''}
+    ${lib.optionalString (!withCloudUi) ''
+      rm -rf $out/share/netdata/web/index.html
+      cp $out/share/netdata/web/v1/index.html $out/share/netdata/web/index.html
+    ''}
   '';

```

RaitoBezarius committed:
netdata: fix v1 dashboard installation

Co-authored-by: Izorkin
Change-Id: I5a379c8622323d7db4fe01f462627d941687ce22
Signed-off-by: Raito Bezarius <masterancpp@gmail.com>

RaitoBezarius commented:
> Patch ro enable v1 UI if disabled CloudUI:
> 
> ```diff
> diff --git a/pkgs/tools/system/netdata/default.nix b/pkgs/tools/system/netdata/default.nix
> index 7fae2c9c8247..9ddba232de19 100644
> --- a/pkgs/tools/system/netdata/default.nix
> +++ b/pkgs/tools/system/netdata/default.nix
> @@ -113,6 +113,10 @@ stdenv.mkDerivation rec {
>        mv $out/libexec/netdata/plugins.d/network-viewer.plugin \
>           $out/libexec/netdata/plugins.d/network-viewer.plugin.org
>      ''}
> +    ${lib.optionalString (!withCloudUi) ''
> +      rm -rf $out/share/netdata/web/index.html
> +      cp $out/share/netdata/web/v1/index.html $out/share/netdata/web/index.html
> +    ''}
>    '';
> ```

Done.

rhoriguchi commented:
Small fix for cups plugin #319032 without `cups.dev` the plugin will not be built even if the option `ENABLE_PLUGIN_CUPS` is enabled

https://github.com/NixOS/nixpkgs/blob/a987a20ea01a7b9bea6704a6601008029945059b/pkgs/tools/system/netdata/default.nix#L44-L45

RaitoBezarius commented:
I leave the rest in the hands of those who care about this package.
